docs/book/how-to/manage-the-zenml-server/best-practices-upgrading-zenml.md

```diff
@@ -0,0 +1,54 @@
+    ![Server Migration Step 1](../../.gitbook/assets/server_migration_1.png)
+
+    ![Server Migration Step 2](../../.gitbook/assets/server_migration_2.png)
+
+- **Team Coordination**: If multiple teams share a ZenML server instance, coordinate the upgrade timing to minimize disruption.
+- **Separate Tenants**: Coordination between teams might be difficult if one team requires new features but the other can't upgrade yet. In such cases, it is recommended to use dedicated ZenML server instances per team or product to allow for more flexible upgrade schedules.
+
+### Minimizing Downtime
+
+- **Upgrade Timing**: Plan upgrades during low-activity periods to minimize disruption.
+
+- **Avoid Mid-Pipeline Upgrades**: Be cautious of automated upgrades or redeployments that might interrupt long-running pipelines.
+
+## Upgrading your code
+
+Sometimes, you might have to upgrade your code to work with a new version of ZenML. This is true especially when you are moving from a really old version to a new major version. The following tips might help, in addition to everything you've learnt in this document so far.
+
+### Testing and Compatibility
+
+- **Local Testing**: It's a good idea to update your local server first and run some old pipelines to check for compatibility issues between the old and new versions.
+- **End-to-End Testing**: You can also develop simple end-to-end tests to ensure that the new version works with your pipeline code and your stack. ZenML already has an [extensive test suite](https://github.com/zenml-io/zenml/tree/main/tests) that we use for releases and you can use it as an example.
+- **Artifact Compatibility**: Be cautious with pickle-based [materializers](../../how-to/handle-data-artifacts/handle-custom-data-types.md), as they can be sensitive to changes in Python versions or libraries. Consider using version-agnostic materialization methods for critical artifacts.
+
+### Dependency Management
+
+- **Python Version**: Make sure that the Python version you are using is compatible with the ZenML version you are upgrading to.
+- **External Dependencies**: Be mindful of external dependencies (e.g. from integrations) that might be incompatible with the new version of ZenML. This could be the case when some older versions are no longer supported or maintained and the ZenML integration is updated to use a newer version.
+
+### Handling API Changes
+
+While ZenML strives for backward compatibility, be prepared for occasional breaking changes (e.g., the Pydantic 2 upgrade).
+
+- **Changelog Review**: Always review the [changelog from new releases](https://github.com/zenml-io/zenml/releases) for new syntax, instructions, or breaking changes.
+- **Migration Scripts**: Use provided [migration scripts](../../how-to/manage-the-zenml-server/migration-guide/migration-guide.md) when available to handle database schema changes.
```

Comment:
this is maybe a bit confusing. this only applies to really old versions?

Reply:
IMO he linked to the right thing and said if it applies. Makes sense to me